### ISSUE TITLE
BUG: <Please write a comprehensive title after the 'BUG: ' prefix>

### ISSUE THREAD

**Issue by unalmis** [scipy/scipy#20525]
### Describe your issue.

ellipeinc and ellipkinc are broken.

### Reproducing Code Example

```python
# Pick some k between 0 and 1.
ellipkinc(np.arcsin(k), 1 / k**2)
ellipeinc(np.arcsin(k), 1 / k**2)
```

### Error message

```shell
nan output is incorrect.
```

### SciPy/NumPy/Python version and system information

```shell
1.12.0 1.26.4 sys.version_info(major=3, minor=12, micro=2, releaselevel='final', serial=0)
Build Dependencies:
  blas:
    detection method: pkgconfig
    found: true
    include directory: /usr/local/include
    lib directory: /usr/local/lib
    name: openblas
    openblas configuration: USE_64BITINT= DYNAMIC_ARCH=1 DYNAMIC_OLDER= NO_CBLAS=
      NO_LAPACK= NO_LAPACKE= NO_AFFINITY=1 USE_OPENMP= HASWELL MAX_THREADS=2
    pc file directory: /usr/local/lib/pkgconfig
    version: 0.3.21.dev
  lapack:
    detection method: pkgconfig
    found: true
    include directory: /usr/local/include
    lib directory: /usr/local/lib
    name: openblas
    openblas configuration: USE_64BITINT= DYNAMIC_ARCH=1 DYNAMIC_OLDER= NO_CBLAS=
      NO_LAPACK= NO_LAPACKE= NO_AFFINITY=1 USE_OPENMP= HASWELL MAX_THREADS=2
    pc file directory: /usr/local/lib/pkgconfig
    version: 0.3.21.dev
  pybind11:
    detection method: config-tool
    include directory: unknown
    name: pybind11
    version: 2.11.1
Compilers:
  c:
    commands: cc
    linker: ld.bfd
    name: gcc
    version: 10.2.1
  c++:
    commands: c++
    linker: ld.bfd
    name: gcc
    version: 10.2.1
  cython:
    commands: cython
    linker: cython
    name: cython
    version: 3.0.8
  fortran:
    commands: gfortran
    linker: ld.bfd
    name: gcc
    version: 10.2.1
  pythran:
    include directory: ../../tmp/pip-build-env-tfqmvy7q/overlay/lib/python3.12/site-packages/pythran
    version: 0.15.0
Machine Information:
  build:
    cpu: x86_64
    endian: little
    family: x86_64
    system: linux
  cross-compiled: false
  host:
    cpu: x86_64
    endian: little
    family: x86_64
    system: linux
Python Information:
  path: /opt/python/cp312-cp312/bin/python
  version: '3.12
```

**Comment by izaid** [comment]
Thanks for reporting this. Actually, like https://github.com/scipy/scipy/issues/20506, we haven't touched the elliptic integrals at all (yet). But they are going to be updated via https://github.com/scipy/scipy/pull/20390. Once that is merged, I think we can look into this easily.

**Comment by cjcopi** [comment]
This bug is the same as I have recently encountered and comes from an explicit check in the incomplete elliptic integrals that is overly strict.

The functions are called, for example, as `ellipkinc(phi, m)`. The code explicitly returns `nan` if  `m>1` (e.g., line 166 in ellik.h). However, the integrals are well defined for all $m < 1 / \sin^2(\phi)$.

For example `ellipkinc(0.015, 1.1)` returns `nan`, however `quad(lambda t: 1 / np.sqrt(1 - 1.1 * np.sin(t)**2), 0, 0.015)` correctly returns `0.015000618791072898`. This value is consistent with the functions `elliptic_f` from `sympy` and  `gsl_sf_ellint_F()` from `libgsl`, both of which handle this case without error.

**Comment by lucascolley** [comment]
cc @steppi now that gh-20390 is in

**Comment by mdhaber** [comment]
Here, the domain is overly restrictive, which could just be considered a documentation oversight and enhancement request, so I will update the label. Note that Boost is even more restrictive with the domain of this function, according to its documentation.

That said, l'll i look into this shortly as a follow-up to https://github.com/scipy/scipy/issues/4452#issuecomment-2323062848. `ellipe` probably has the same feature. In the meantime, see how you might implement this in terms of `elliprf` yourself following Boost's implementation notes at the end of [their documentation](https://live.boost.org/doc/libs/1_38_0/libs/math/doc/sf_and_dist/html/math_toolkit/special/ellint/ellint_1.html).

**Comment by mdhaber** [comment]
Someone more familiar with the existing implementations of the special functions may have better luck, but I wasn't able to simply remove that condition to extend the usable domain. In case that doesn't work:

Assuming the output of this function is real and the domain of $m$ is extended, the domain of $\phi$ is $-\phi_c < \phi < \phi_c$ if $m > 1$, where $\phi_c = \sin^{-1} \sqrt{ \frac{1}{m} }$. (Outside of that, the output would be complex, and implementing that is a different sort of enhancement.) In this region, `special.ellipkinc` ($F$) can be implemented in terms of `special.elliprf` ($R_F$) as:

$$
F(\phi, m) = \sin \phi \ R_F \left( \cos^2 \phi , 1 - m \ \sin^2 \phi, 1 \right).
$$

The Boost documentation doesn't mention it explicitly, but this assumes $\phi > 0$ and $F(-\phi, m) =  -F(\phi, m)$. Including that logic:

```pythin3
def k(phi, m):
    sign = np.sign(phi)
    phi = np.abs(phi)
    val = np.asarray(np.sin(phi) * Rf(np.cos(phi)**2, 1 - m*np.sin(phi)**2, 1))
    return val * sign
```

Focusing on $m > 1$, we can compare the results of this implementation against `mpmath.ellipf`:

<details>

```python3
import numpy as np
import matplotlib.pyplot as plt
from scipy.special import elliprf as Rf
from mpmath import mp
mp.dps = 50

@np.vectorize
def k_ref(phi, m):
    # in case mpmath has tiny imaginary component
    return complex(mp.ellipf(phi, m)).real

@np.vectorize
def k(phi, m):
    sign = np.sign(phi)
    phi = np.abs(phi)
    val = np.asarray(np.sin(phi) * Rf(np.cos(phi)**2, 1 - m*np.sin(phi)**2, 1))
    return val * sign

def log10_relative_error(res, ref):
    with np.errstate(all='ignore'):
        out = np.where(res != ref, abs(res - ref) / abs(ref), 0)
        out = np.log10(out)
    out[out < -16] = -16
    return out

m = np.logspace(0, 3, 30)
phi_c = np.asin(np.sqrt(1/m))
phi = phi_c * np.linspace(-1, 1, 25)[:, np.newaxis]
m, phi = np.broadcast_arrays(m, phi)

res = k(phi,  m)
ref = k_ref( phi,  m)

logerr = log10_relative_error(res, ref)
plt.contourf(m, phi, logerr, cmap='viridis')
plt.xlabel(r"$m$")
plt.ylabel(r"$\phi$")
plt.colorbar()
plt.title(r"$\log_{10}$ error")
plt.xscale('log')
plt.show()
```

</details>

<img width="863" alt="image" src="https://github.com/user-attachments/assets/8b24470c-6e31-4b72-bd8e-ebb25248ba3b">

Note the the plot disappears for $| \phi | > \phi_c$ and the error is greater at $\phi \approx \phi_c$, but the error is very small   throughout most of the domain. So if simply removing the explicit condition for $m > 1$ doesn't work, the function could branch to follow this implementation instead.

Alternatively, the Cephes implementation could be replaced. For $-1 < m < 1$ and $-\frac{\phi}{2} < \phi < \frac{\phi}{2}$, the error looks like:
<img width="857" alt="image" src="https://github.com/user-attachments/assets/714c44f5-3ca8-4491-a4cb-3b5964d2f63b">

The logic for extending the domain to arbitrary $\phi$ for $m < 1$ could look like the following, although there might be a simpler way to do it:
```python3@np.vectorize
def k_unit(phi, m):
    sign = np.sign(phi)
    phi = np.abs(phi)
    val = np.asarray(np.sin(phi) * Rf(np.cos(phi)**2, 1 - m*np.sin(phi)**2, 1))
    return val * sign

@np.vectorize
def k(phi, m):
    # standardize to positive phi
    sign = np.sign(phi)
    phi = np.abs(phi)

    # shift to [-np.pi, np.pi], but count the number
    # of complete periods
    n = (phi + np.pi/2) // np.pi
    phi = (phi + np.pi/2) % np.pi - np.pi/2

    # Add the appropriate number of complete integrals
    # to the result
    K = k_unit(np.pi/2, m)
    val = k_unit(phi, m)
    return sign*(val + 2*n*K)
```

It looks like the error would be comparable to what we have now for any values of $m$, $\phi$.

**Comment by unalmis** [comment]
> Here, the function does not produce incorrect results; rather, the domain is restricted compared to some versions of the function, and this restriction is not mentioned in the documentation. I've updated the label to reflect that this is really a documentation oversight and enhancement request rather than a "bug". 

I disagree with this. If it's helpful I can give a simple proof to show that both of these integrals, using their common definition, are finite and real-valued for any $k \in (0, 1)$.
```python
ellipkinc(np.arcsin(k), 1 / k**2)
ellipeinc(np.arcsin(k), 1 / k**2)
```

Maybe I have just cherry-picked your comment and we're actually both in agreement, but pointing out in case not.

**Comment by mdhaber** [comment]
I agree that $F(\sin^{-1}k, 1/k^2)$ is real-valued for $k < 1$ . But the implementation `ellipkinc` does not currently include `k < 1` in its domain. Likewise, $F$ is defined in regions where the output would be complex. `ellipkinc` doesn't include those arguments in its domain either. Authors may choose to restrict the domain of their implementation of a function to a region in which they are prepared to accurately perform the calculation and test the results. There are many reasons why the domain of an implementation might not include all regions where a function might be defined. For instance, it may be challenging to perform the calculation numerically in some regions. I'm not sure why the authors the underlying Cephes function chose to restrict the domain here, but whatever it was, it looks intentional. There are many other instances of this in `scipy`, and even the Boost implementation of this function has the same restriction.

That said, It should definitely be documented, and requesting that the domain be extended is an excellent feature request.
